public/app/features/explore/TraceView/components/TracePageHeader/TracePageHeader.tsx

```diff
@@ -198,94 +306,139 @@ export const TracePageHeader = memo((props: TracePageHeaderProps) => {
 
 TracePageHeader.displayName = 'TracePageHeader';
 
-const getNewStyles = (theme: GrafanaTheme2) => {
+const getStyles = (theme: GrafanaTheme2) => {
   return {
-    TracePageHeaderBack: css({
-      label: 'TracePageHeaderBack',
-      alignItems: 'center',
-      alignSelf: 'stretch',
-      backgroundColor: '#fafafa',
-      borderBottom: '1px solid #ddd',
-      borderRight: '1px solid #ddd',
-      color: 'inherit',
-      display: 'flex',
-      fontSize: '1.4rem',
-      padding: '0 1rem',
-      marginBottom: '-1px',
-      '&:hover': {
-        backgroundColor: '#f0f0f0',
-        borderColor: '#ccc',
-      },
-    }),
-    TracePageHeaderOverviewItemValueDetail: cx(
-      css({
-        label: 'TracePageHeaderOverviewItemValueDetail',
-        color: '#aaa',
-      }),
-      'trace-item-value-detail'
-    ),
-    TracePageHeaderOverviewItemValue: css({
-      label: 'TracePageHeaderOverviewItemValue',
-      '&:hover > .trace-item-value-detail': {
-        color: 'unset',
-      },
-    }),
     header: css({
       label: 'TracePageHeader',
       backgroundColor: theme.colors.background.primary,
-      padding: '0.5em 0 0 0',
+      padding: '0.5em',
       position: 'sticky',
       top: 0,
       zIndex: 5,
       textAlign: 'left',
     }),
+
     titleRow: css({
+      display: 'flex',
       alignItems: 'flex-start',
+      justifyContent: 'space-between',
+      marginBottom: theme.spacing(1),
+      gap: theme.spacing(2),
+    }),
+
+    titleSection: css({
       display: 'flex',
-      padding: '0 8px',
-      flexWrap: 'wrap',
+      alignItems: 'center',
+      gap: theme.spacing(2),
+      flex: 1,
+      minWidth: 0, // Allow text truncation
     }),
+
     title: css({
-      color: 'inherit',
-      flex: 1,
-      fontSize: '1.7em',
-      lineHeight: '1em',
-      marginBottom: 0,
-      minWidth: '200px',
+      color: theme.colors.text.primary,
+      fontSize: theme.typography.h3.fontSize,
+      fontWeight: theme.typography.h3.fontWeight,
+      lineHeight: theme.typography.h3.lineHeight,
+      margin: 0,
+      minWidth: 0,
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+      whiteSpace: 'nowrap',
     }),
-    subtitle: css({
-      flex: 1,
-      lineHeight: '1em',
-      margin: '-0.5em 0.5em 0.75em 0.5em',
+
+    badges: css({
+      display: 'flex',
+      gap: theme.spacing(1),
+      alignItems: 'center',
+      flexShrink: 0,
+    }),
+
+    actions: css({
+      display: 'flex',
+      alignItems: 'center',
+      gap: theme.spacing(1),
+      flexShrink: 0,
+    }),
+
+    shareButtonGroup: css({
+      display: 'flex',
+    }),
+
+    shareDropdownButton: css({
+      borderLeft: 'none !important',
+      // eslint-disable-next-line @grafana/no-border-radius-literal
+      borderTopLeftRadius: '0 !important',
+      // eslint-disable-next-line @grafana/no-border-radius-literal
+      borderBottomLeftRadius: '0 !important',
+      marginLeft: '-1px',
```

Comment:
I think you may not need tweaking the css if you wrap the button and dropdown in `<ButtonGroup>`. We do it in [Explore's share button](https://github.com/grafana/grafana/blob/20c700dd52373c7b4e6f9560d0c2e35985308cfb/public/app/features/explore/ShortLinkButtonMenu.tsx#L133)